src/ScriptCs.Hosting/ScriptServicesBuilder.cs

```diff
@@ -131,6 +133,18 @@ public IScriptServicesBuilder Debug(bool debug = true)
             return this;
         }
 
+        public IScriptServicesBuilder SetOverride<TKey, TValue>(TValue value) where TValue : TKey
```

Comment:
As Key needs to be a contract, I'd suggest `SetOverride<TContract, TImpl>(TImpl value) where TImpl : TContract`